### PR TITLE
Add support for avc1 and avc3 stream formats and remove H264.RemoteStream

### PR DESCRIPTION
**Pull request by Noarkhh**
- Add `stream_structure` field and type to H264 that describes the format of the stream as either annexb, avc1 (with DCR) or avc3 (with DCR).
- Remove `H264.RemoteStream` stream format and give it's role of describing unparsed or DCR having H264 stream to H264 stream format.

### CODE REVIEW

File: lib/membrane_h264_format/h264.ex [hunk 1]
```diff
@@ -38,24 +40,42 @@ defmodule Membrane.H264 do
...
+  - `unprefixed_poslen: {pos, len}` - as above, but omits Annex B or AVC prefix
   - `metadata: metadata` - metadata that would be merged into the buffer metadata
-    if `alignment` was `:nal`.
+    if `alignment` was `:nalu`.
   """
-  @type nalu_in_metadata_t :: boolean()
+  @type nalu_in_metadata :: boolean()
+
+  @typedoc """
+  Describes h264 stream structure.
```
Comment (varsill) [inline lib/membrane_h264_format/h264.ex]:
```suggestion
  Describes H264 stream structure.
```

Comment (varsill) [reply lib/membrane_h264_format/h264.ex]: I guess this is the correct styling, we can also change the other occurrences of `h264` into `H264` :D

File: lib/membrane_h264_format/h264.ex [hunk 2]
```diff
@@ -38,24 +40,42 @@ defmodule Membrane.H264 do
...
+
+  Annex B ([ITU-T H.264 Recommendation](http://www.itu.int/rec/T-REC-H.264-201704-I/en))
+  is suitable for writing to file or streaming with MPEG-TS.
+  In this format each NAL unit is prefixed by three or four-byte start code (`0x(00)000001`)
+  that allows to identify boundaries.
+
+  avc1 and avc3 are described by ISO/IEC 14496-15. In such stream a DCR (Decoder Configuration
+  Record) is included as metadata and NALUs lack the start codes, but are prefixed with their length.
+  The length of these prefixes is contained in the stream's DCR Avc streams are more suitable for
```
Comment (varsill) [inline lib/membrane_h264_format/h264.ex]:
```suggestion
  The length of these prefixes is contained in the stream's DCR. AVC streams are more suitable for
```

File: lib/membrane_h264_format/h264.ex [hunk 3]
```diff
@@ -38,24 +40,42 @@ defmodule Membrane.H264 do
...
+  placing in containers (e.g. they are used by QuickTime (.mov), MP4, Matroska and FLV).
+  Additionally, the stream's. In avc1 streams PPSs and SPSs (Picture Parameter Sets and
+  Sequence Parameter Sets) are transported in the DCR, when in avc3 they may be also present in
+  the stream (in-band).
```
Comment (varsill) [inline lib/membrane_h264_format/h264.ex]: There is something missing there ;)

**Review by varsill — CHANGES_REQUESTED** [review]

File: lib/membrane_h264_format/h264.ex (hunk 3 above)
Comment (Noarkhh) [reply lib/membrane_h264_format/h264.ex]: yes, forgot to push

**Review by varsill — APPROVED** [review]

**Pull request merged**